sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectSessionManager.scala

```diff
@@ -176,21 +173,24 @@ class SparkConnectSessionManager extends Logging {
     sessionHolder.foreach(shutdownSessionHolder(_))
   }
 
-  private[connect] def shutdown(): Unit = sessionsLock.synchronized {
-    scheduledExecutor.foreach { executor =>
-      ThreadUtils.shutdown(executor, FiniteDuration(1, TimeUnit.MINUTES))
+  private[connect] def shutdown(): Unit = {
+    sessionsLock.synchronized {
```

Comment:
Can you add Julek's comment here on why sessionLock is needed here?

Reply:
Thanks for reviewing this! As far as I understand, his comments address a potential data race between sessionStore and closedSessionsCache, and the data race can be resolved without relying on this particular lock: not quite related to this piece of code.

Locking here is to protect scheduledExecutor, and I think that is self-explanatory at line 55 (@GuardedBy("sessionsLock")).

Reply:
Ah I see the comment above 👍